### DISCUSSION TITLE
Array or List type for collection field

### DISCUSSION THREAD

**Discussion by Aragur** [pocketbase/pocketbase#1347] [Q&A] [answered]
Is it somehow possible to save an array of strings in a collection?
I cannot find any types for that in the docs.
There is a select (multiple) field which neatly returns an array via the API endpoint.
I want to add a custom string array, is this somehow possible?

**Comment by ganigeorgiev** [comment, marked as answer]
The `select` field type works only with predefined list of values. If you want arbitrary array values, you can:

- use the `json` field type (there are no data validations)

- use a `relation` field to another collection (this could be helpful if your values are object-like, aka. have more than one string field)

---

Also, please note that currently the multi `relation` field type has 2 limitations:
- There is no multi-match filter support yet and when attempting to filter a multi-value field it will return `true` on the first match (eg. `someRelField.status = 'active'` will return true if there is at least one `someRelField` record that match the condition).
- When submitting/removing items you have to always pass the final array, eg:
   ```js
   await pb.collect('example').update('RECORD_ID', { someRelField: ['id1', 'id2', 'id3'] })
   ```
These 2 limitations will be fixed with the next release soon.

**Reply by Aragur** [reply]
@ganigeorgiev Thanks a lot :) using `json`directly parses it in the JS-SDK, that's a really nice feature!
